### PR TITLE
reference: update docs to reflect changes in `tidb/config/config.go`

### PR DESCRIPTION
**Pull request by ichn-hu**
According to pingcap/tidb#12853

### What is changed, added or deleted? 

Update `disable-timestamp` to `enable-timestamp`.

### What is the related PR or file link(s)? 

https://github.com/pingcap/tidb/pull/12853/

### Which version does your change affect? 

dev

### CODE REVIEW

**Comment by ichn-hu** [comment]
@yikeke Hi, what should I do to get this PR merged?

**Comment by yikeke** [comment]
> @yikeke Hi, what should I do to get this PR merged?

Hi @ichn-hu you need to ask for a technical review from the engineering team and a language/format review from the I18N team. When you get two LGTM (Looks Good To Me), then you can merge this pull request.

**Comment by yikeke** [comment]
`dev` is the latest version. Did your changes also apply to `v3.0` or `v2.1`? @ichn-hu If yes, you also need to update the `v3.0` or `v2.1` file accordingly.

**Comment by ichn-hu** [comment]
> `dev` is the latest version. Did your changes also apply to `v3.0` or `v2.1`? @ichn-hu If yes, you also need to update the `v3.0` or `v2.1` file accordingly.
> 
> Rest LGTM

It is only applied to the current master branch, and will be effective in next GA, therefore dev is affected.

And the change is back-compatible, `disble-xx` is still supported so no need to change anything in previous documentation.

File: dev/reference/configuration/tidb-server/configuration-file.md [hunk 1]
```diff
@@ -74,11 +74,11 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
 
-### `disable-timestamp`
+### `enable-timestamp`
 
-+ 是否禁止在日志中输出时间戳。
-+ 默认值：false
-+ 如果设置为 true，那么日志里面将不会输出时间戳。
++ 是否在日志中输出时间戳。
++ 默认值：true
++ 如果设置为 false，那么日志里面将不会输出时间戳。
```
Comment (zz-jason) [inline dev/reference/configuration/tidb-server/configuration-file.md]: 可以加个注释：

注：考虑后向兼容性，原来的配置项 `disable-timestamp` 仍然有效，但是如果和 `enable-timestamp` 配置的值在语义上冲突，TiDB 将按照 xx 的行为处理。请废弃 `disable-timestamp` 的用法，使用语义上更易于理解的 `enable-timestamp`。

Comment (ichn-hu) [reply dev/reference/configuration/tidb-server/configuration-file.md]: 好的！

**Comment by ichn-hu** [comment]
@zz-jason Note added, PTAL.

**Comment by lilin90** [comment]
@ichn-hu So this change doesn't apply to v3.1 too?

**Comment by ichn-hu** [comment]
> @ichn-hu So this change doesn't apply to v3.1 either?

No, it does not. I've checked the code in release 3.1 branch, my commit is not included.

**Comment by lilin90** [comment]
@ichn-hu Got it. Thanks for your check.

File: dev/reference/configuration/tidb-server/configuration-file.md [hunk 2]
```diff
@@ -74,11 +74,15 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
+
+> **注意：**
+>
+> 考虑后向兼容性，原来的配置项 `disable-timestamp` 仍然有效，但如果和 `enable-timestamp` 配置的值在语义上冲突（譬如在配置中把 `enable-timestamp` 和 `disable-timestamp` 同时设置为 `true`）则 TiDB 会报错并拒绝启动。请废弃 `disable-timestamp` 的用法，使用语义上更易于理解的 `enable-timestamp`。
```
Comment (lilin90) [inline dev/reference/configuration/tidb-server/configuration-file.md]:
```suggestion
> 考虑后向兼容性，原来的配置项 `disable-timestamp` 仍然有效，但如果和 `enable-timestamp` 配置的值在语义上冲突（例如在配置中把 `enable-timestamp` 和 `disable-timestamp` 同时设置为 `true`），则 TiDB 会报错并拒绝启动。请废弃 `disable-timestamp` 的用法，使用语义上更易于理解的 `enable-timestamp`。
```

**Comment by lilin90** [comment]
@zz-jason PTAL again~ Thanks!

**Comment by zz-jason** [comment]
@ichn-hu Should we continue to review this PR until your PR in tidb repo is merged?

**Comment by ichn-hu** [comment]
@zz-jason , I would like to see this PR merged first. I still stand with the current behavior, the new PR in TiDB needs more thorough discussion, so for now let's just merge this PR to reflect the changes already there in the TiDB repo. If the new PR is decided to get merged, I could send another PR in docs-cn.

**Comment by zz-jason** [comment]
> @zz-jason , I would like to see this PR merged first. I still stand with the current behavior, the new PR in TiDB needs more thorough discussion, so for now let's just merge this PR to reflect the changes already there in the TiDB repo. If the new PR is decided to get merged, I could send another PR in docs-cn.

OK

**Pull request merged**